### ISSUE TITLE
No filltering happnes per Terraform category

### ISSUE THREAD

**Issue by aabouzaid** [stelligent/config-lint#165]
Hello,

When no resource specified in Terraform lint, it loads all categorises and emit error.

Example:

```shell
cat << EOF > tf-rule.yml
---
version: 1
description: Terraform module invocation example
type: Terraform
files:
  - "*.tf"
rules:
  - id: MODULE_EXAMPLE
    category: resource
    assertions:
      - key: ami
        op: present
EOF
```

With this tf file:

```shell
cat << EOF > main.tf
resource "aws_instance" "web" {
  ami           = "ami-a1b2c3d4"
  instance_type = "t2.micro"
}

data "aws_ami" "example" {
  most_recent = true

  owners = ["self"]
  tags = {
    Name   = "app-server"
    Tested = "true"
  }
}

provider "google" {
  project = "acme-app"
  region  = "us-central1"
}
EOF
```
Config-lint tries to apply the rule to all block types and emit this error (tested with tf12 and tf11 parsers):
```
$ docker run --rm -w /data -v $(pwd):/data stelligent/config-lint -tfparser tf12 -rules tf-rule.yml main.tf

[
  {
    "AssertionMessage": "ami should be present",
    "Category": "data",
    "CreatedAt": "2020-03-29T15:56:32Z",
    "Filename": "main.tf",
    "LineNumber": 0,
    "ResourceID": "example",
    "ResourceType": "aws_ami",
    "RuleID": "MODULE_EXAMPLE",
    "RuleMessage": "",
    "Status": "FAILURE"
  },
  {
    "AssertionMessage": "ami should be present",
    "Category": "provider",
    "CreatedAt": "2020-03-29T15:56:32Z",
    "Filename": "main.tf",
    "LineNumber": 0,
    "ResourceID": "1",
    "ResourceType": "google",
    "RuleID": "MODULE_EXAMPLE",
    "RuleMessage": "",
    "Status": "FAILURE"
  }
]
```

What's expected, when there is no resource, the rule should be limited to the category.

**Comment by aabouzaid** [comment]
Ah sorry, looks like it's duplicated #135.

**Comment by aabouzaid** [comment]
Fixed by #167
